tests/test_oneclick.py

```diff
@@ -0,0 +1,175 @@
+import datetime
+import json
+from lxml import etree
+from nose.tools import (
+    eq_, 
+    assert_raises,
+    assert_raises_regexp,
+    set_trace,
+)
+import os
+from StringIO import StringIO
+
+
+from core.model import (
+    DataSource,
+    Edition,
+    Identifier,
+    Subject,
+    Contributor,
+    LicensePool,
+)
+
+from core.metadata_layer import (
+    Metadata,
+    CirculationData,
+    IdentifierData,
+    ContributorData,
+    SubjectData,
+)
+
+
+from api.oneclick import (
+    OneClickAPI,
+    MockOneClickAPI,
+)
+
+
+from . import (
+    DatabaseTest,
+    #sample_data
+)
+
+'''
+from api.circulation import (
+    LoanInfo,
+    HoldInfo,
+    FulfillmentInfo,
+)
+
+from api.circulation_exceptions import *
+
+from core.analytics import Analytics
+'''
+
+
+class OneClickAPITest(DatabaseTest):
+
+    def setup(self, _db=None):
+        super(OneClickAPITest, self).setup()
+        self.api = MockOneClickAPI(self._db)
+        #self.api = OneClickAPI(self._db)
+        base_path = os.path.split(__file__)[0]
+        self.resource_path = os.path.join(base_path, "files", "oneclick")
+
+
+    def get_data(self, filename):
+        # returns contents of sample file as string and as dict
+        path = os.path.join(self.resource_path, filename)
+        data = open(path).read()
+        return data, json.loads(data)
+
+
+
+class TestOneClickAPI(OneClickAPITest):
+
+    def test_get_patron_internal_id(self):
+        datastr, datadict = self.get_data("response_patron_internal_id_not_found.json")
+        self.api.queue_response(status_code=200, content=datastr)
+        oneclick_patron_id = self.api.get_patron_internal_id(patron_cardno='9305722621')
+        eq_(None, oneclick_patron_id)
+
+        datastr, datadict = self.get_data("response_patron_internal_id_error.json")
+        self.api.queue_response(status_code=500, content=datastr)
+        assert_raises_regexp(
+            ValueError, "Got status code 500 from external server, cannot continue.", 
+            self.api.get_patron_internal_id, patron_cardno='130572262x'
+        )
```

Comment:
It's for the third-party server breaking. It means we can't fulfill the patron's request (or do whatever we were trying to do), but there's absolutely nothing we can do on our end to fix the problem except bother the third party about it.